lib/util/language_utils.js

```diff
@@ -455,45 +455,209 @@ shaka.util.LanguageUtils = class {
-  ['kik', 'ki'], ['kin', 'rw'], ['kir', 'ky'], ['kom', 'kv'], ['kon', 'kg'],
-  ['kor', 'ko'], ['kua', 'kj'], ['kur', 'ku'], ['lao', 'lo'], ['lat', 'la'],
-  ['lav', 'lv'], ['lim', 'li'], ['lin', 'ln'], ['lit', 'lt'], ['ltz', 'lb'],
-  ['lub', 'lu'], ['lug', 'lg'], ['mac', 'mk'], ['mah', 'mh'], ['mal', 'ml'],
-  ['mao', 'mi'], ['mar', 'mr'], ['may', 'ms'], ['mkd', 'mk'], ['mlg', 'mg'],
-  ['mlt', 'mt'], ['mon', 'mn'], ['mri', 'mi'], ['msa', 'ms'], ['mya', 'my'],
-  ['nau', 'na'], ['nav', 'nv'], ['nbl', 'nr'], ['nde', 'nd'], ['ndo', 'ng'],
-  ['nep', 'ne'], ['nld', 'nl'], ['nno', 'nn'], ['nob', 'nb'], ['nor', 'no'],
-  ['nya', 'ny'], ['oci', 'oc'], ['oji', 'oj'], ['ori', 'or'], ['orm', 'om'],
-  ['oss', 'os'], ['pan', 'pa'], ['per', 'fa'], ['pli', 'pi'], ['pol', 'pl'],
-  ['por', 'pt'], ['pus', 'ps'], ['que', 'qu'], ['roh', 'rm'], ['ron', 'ro'],
-  ['rum', 'ro'], ['run', 'rn'], ['rus', 'ru'], ['sag', 'sg'], ['san', 'sa'],
-  ['sin', 'si'], ['slk', 'sk'], ['slo', 'sk'], ['slv', 'sl'], ['sme', 'se'],
-  ['smo', 'sm'], ['sna', 'sn'], ['snd', 'sd'], ['som', 'so'], ['sot', 'st'],
-  ['spa', 'es'], ['sqi', 'sq'], ['srd', 'sc'], ['srp', 'sr'], ['ssw', 'ss'],
-  ['sun', 'su'], ['swa', 'sw'], ['swe', 'sv'], ['tah', 'ty'], ['tam', 'ta'],
-  ['tat', 'tt'], ['tel', 'te'], ['tgk', 'tg'], ['tgl', 'tl'], ['tha', 'th'],
-  ['tib', 'bo'], ['tir', 'ti'], ['ton', 'to'], ['tsn', 'tn'], ['tso', 'ts'],
-  ['tuk', 'tk'], ['tur', 'tr'], ['twi', 'tw'], ['uig', 'ug'], ['ukr', 'uk'],
-  ['urd', 'ur'], ['uzb', 'uz'], ['ven', 've'], ['vie', 'vi'], ['vol', 'vo'],
-  ['wel', 'cy'], ['wln', 'wa'], ['wol', 'wo'], ['xho', 'xh'], ['yid', 'yi'],
-  ['yor', 'yo'], ['zha', 'za'], ['zho', 'zh'], ['zul', 'zu'],
+  ['aar', 'aa'],
+  ['abk', 'ab'],
+  ['afr', 'af'],
+  ['aka', 'ak'],
+  ['alb', 'sq'],
+  ['amh', 'am'],
+  ['ara', 'ar'],
+  ['arg', 'an'],
+  ['arm', 'hy'],
+  ['asm', 'as'],
+  ['ava', 'av'],
+  ['ave', 'ae'],
+  ['aym', 'ay'],
+  ['aze', 'az'],
+  ['bak', 'ba'],
+  ['bam', 'bm'],
+  ['baq', 'eu'],
+  ['bel', 'be'],
+  ['ben', 'bn'],
+  ['bih', 'bh'],
+  ['bis', 'bi'],
+  ['bod', 'bo'],
+  ['bos', 'bs'],
+  ['bre', 'br'],
+  ['bul', 'bg'],
+  ['bur', 'my'],
+  ['cat', 'ca'],
+  ['ces', 'cs'],
+  ['cha', 'ch'],
+  ['che', 'ce'],
+  ['chi', 'zh'],
+  ['chu', 'cu'],
+  ['chv', 'cv'],
+  ['cor', 'kw'],
+  ['cos', 'co'],
+  ['cre', 'cr'],
+  ['cym', 'cy'],
+  ['cze', 'cs'],
+  ['dan', 'da'],
+  ['deu', 'de'],
+  ['div', 'dv'],
+  ['dut', 'nl'],
+  ['dzo', 'dz'],
+  ['ell', 'el'],
+  ['eng', 'en'],
+  ['epo', 'eo'],
+  ['est', 'et'],
+  ['eus', 'eu'],
+  ['ewe', 'ee'],
+  ['fao', 'fo'],
+  ['fas', 'fa'],
+  ['fij', 'fj'],
+  ['fin', 'fi'],
+  ['fra', 'fr'],
+  ['fre', 'fr'],
+  ['fry', 'fy'],
+  ['ful', 'ff'],
+  ['geo', 'ka'],
+  ['ger', 'de'],
+  ['gla', 'gd'],
+  ['gle', 'ga'],
+  ['glg', 'gl'],
+  ['glv', 'gv'],
+  ['gre', 'el'],
+  ['grn', 'gn'],
+  ['guj', 'gu'],
+  ['hat', 'ht'],
+  ['hau', 'ha'],
+  ['heb', 'he'],
+  ['heb', 'iw'],
```

Comment:
It's a map after all, so only `heb -> iw` mapping will be used. Is that what we want?

Reply:
Nope!  Not at all.  Good catch.

I'll restructure this.

Reply:
Oh, wait a sec, I see the root of my confusion now.

In Shaka Player, we only ever map from 3-letter codes to 2-letter codes, to canonicalize inputs for comparison.  So this extra mapping is not needed.

In Shaka Packager, we initially copied the map from Player.  But Packager uses it in both directions, because to create an mp4, you need the 3-letter code in all cases.  So one input with "he" and another with "iw" both need to map to "heb" when Packager creates the MP4.  But then when Packager creates the MPD, it needs to follow BCP-47 and map down to the shortest possible tag.

There's an edge case here for the player, where one stream says "he" and another says "iw", and the player would currently treat them as different.  This could also happen for other languages with multiple 2-letter codes.  However, only one 2-letter code is canonical.  In this case, "he" is the official ISO-639-2 code for Hebrew, whereas "iw" was used before standardization and still appears in some places.  So I would argue that we could ignore this edge case, and leave it to packaging software to create DASH manifests with canonical language codes.

With that, I'm closing this PR.